models/base/scratch/bigquery/snowplow_ecommerce_base_events_this_run.sql

```diff
@@ -21,61 +21,33 @@ from (
-        fields=user_fields(),
-        col_prefix='contexts_com_snowplowanalytics_snowplow_ecommerce_user_1_',
-        relation=ref('snowplow_ecommerce_events_stg'),
-        relation_alias='a') }},
-
-    -- unpacking the ecommerce checkout step object
-    {{ snowplow_utils.get_optional_fields(
-        enabled=not var('snowplow__disable_ecommerce_checkouts', false),
-        fields=checkout_step_fields(),
-        col_prefix='contexts_com_snowplowanalytics_snowplow_ecommerce_checkout_step_1_',
-        relation=ref('snowplow_ecommerce_events_stg'),
-        relation_alias='a') }},
-
-    -- unpacking the ecommerce page object
-    {{ snowplow_utils.get_optional_fields(
-        enabled=not var('snowplow__disable_ecommerce_page_context', false),
-        fields=tracking_page_fields(),
-        col_prefix='contexts_com_snowplowanalytics_snowplow_ecommerce_page_1_',
-        relation=ref('snowplow_ecommerce_events_stg'),
-        relation_alias='a') }},
-
-    -- unpacking the ecommerce transaction object
-    {{ snowplow_utils.get_optional_fields(
-        enabled=not var('snowplow__disable_ecommerce_transactions', false),
-        fields=transaction_fields(),
-        col_prefix='contexts_com_snowplowanalytics_snowplow_ecommerce_transaction_1_',
-        relation=ref('snowplow_ecommerce_events_stg'),
-        relation_alias='a') }},
-
-    -- unpacking the ecommerce cart object
-    {{ snowplow_utils.get_optional_fields(
-        enabled=not var('snowplow__disable_ecommerce_carts', false),
-        fields=cart_fields(),
-        col_prefix='contexts_com_snowplowanalytics_snowplow_ecommerce_cart_1_',
-        relation=ref('snowplow_ecommerce_events_stg'),
-        relation_alias='a') }},
-
-    -- unpacking the ecommerce action object
-    {{ snowplow_utils.get_optional_fields(
-        enabled=true,
-        fields=tracking_action_fields(),
-        col_prefix='unstruct_event_com_snowplowanalytics_snowplow_ecommerce_snowplow_ecommerce_action_1_',
-        relation=ref('snowplow_ecommerce_events_stg'),
-        relation_alias='a') }},
+      {% if var('snowplow__enable_mobile_events', false) %}
+        coalesce(
+          {{ snowplow_utils.get_optional_fields(
+          enabled=var('snowplow__enable_mobile_events', false),
+          fields=[{'field': 'id', 'dtype': 'string'}],
+          col_prefix='contexts_com_snowplowanalytics_mobile_screen_1_',
+          relation=source('atomic', 'events'),
+          relation_alias='a',
+          include_field_alias=false) }},
+          a.contexts_com_snowplowanalytics_snowplow_web_page_1_0_0[safe_offset(0)].id
+        ) as page_view_id,
+        coalesce(
+          {{ snowplow_utils.get_optional_fields(
+          enabled=var('snowplow__enable_mobile_events', false),
+          fields=[{'field': 'session_id', 'dtype': 'string'}],
+          col_prefix='contexts_com_snowplowanalytics_snowplow_client_session_1_',
+          relation=source('atomic', 'events'),
```

Comment:
I tested overwriting the schema and tablename as well and worked fine as expected.